pkg/bicep/biceptoschema.go

```diff
@@ -0,0 +1,231 @@
+		return errors.New("unknown type: " + bicepParam.TypeString)
+	}
+}
+
+func parseIntParam(sch *schema.Schema, bicepParam bicepParam) error {
+	sch.Type = "integer"
+	sch.Default = bicepParam.DefaultValue
+
+	if bicepParam.AllowedValues != nil && len(bicepParam.AllowedValues) == 1 {
+		sch.Enum = bicepParam.AllowedValues[0].([]interface{})
+	}
+
+	if bicepParam.MinValue != nil {
+		sch.Minimum = json.Number(fmt.Sprintf("%d", *bicepParam.MinValue))
+	}
+	if bicepParam.MaxValue != nil {
+		sch.Maximum = json.Number(fmt.Sprintf("%d", *bicepParam.MaxValue))
+	}
+
+	return nil
+}
+
+func parseBoolParam(sch *schema.Schema, bicepParam bicepParam) error {
+	sch.Type = "boolean"
+	sch.Default = bicepParam.DefaultValue
+	return nil
+}
+
+func parseStringParam(sch *schema.Schema, bicepParam bicepParam, secure bool) error {
+	sch.Type = "string"
+	sch.Default = bicepParam.DefaultValue
+
+	if secure {
+		sch.Format = "password"
+	}
+
+	if bicepParam.AllowedValues != nil && len(bicepParam.AllowedValues) == 1 {
+		sch.Enum = bicepParam.AllowedValues[0].([]interface{})
+	}
+
+	sch.MinLength = bicepParam.MinLength
+	sch.MaxLength = bicepParam.MaxLength
+
+	return nil
+}
+
+func parseArrayParam(sch *schema.Schema, bicepParam bicepParam) error {
+	sch.Type = "array"
+
+	sch.MinItems = bicepParam.MinLength
+	sch.MaxItems = bicepParam.MaxLength
+
+	if bicepParam.DefaultValue != nil && len(bicepParam.DefaultValue.([]interface{})) != 0 {
+		parseArrayType(sch, bicepParam.DefaultValue.([]interface{}))
+	}
+	return nil
+}
+
+func parseObjectParam(sch *schema.Schema, bicepParam bicepParam) error {
+	sch.Type = "object"
+
+	if bicepParam.DefaultValue != nil && len(bicepParam.DefaultValue.(map[string]interface{})) > 1 {
+		parseObjectType(sch, bicepParam.DefaultValue.(map[string]interface{}))
+	}
+	return nil
+}
+
+func parseObjectType(sch *schema.Schema, objValue map[string]interface{}) error {
+	sch.Properties = orderedmap.New[string, *schema.Schema]()
+	sch.Required = []string{}
+
+	for name, value := range objValue {
+		if name == "_kics_lines" {
+			continue
+		}
+
+		property := new(schema.Schema)
+		property.Title = name
+
+		switch reflect.TypeOf(value).Kind() {
+		case reflect.Float64:
+			property.Type = "integer"
+			property.Default = value
+		case reflect.Bool:
+			property.Type = "boolean"
+			property.Default = value
+		case reflect.String:
+			property.Type = "string"
+			property.Default = value
+		case reflect.Slice:
+			property.Type = "array"
+			parseArrayType(property, value.([]interface{}))
+		case reflect.Map:
+			property.Type = "object"
+			parseObjectType(property, value.(map[string]interface{}))
+		default:
+			return errors.New("unknown type: " + reflect.TypeOf(value).String())
+		}
+
+		sch.Properties.Set(name, property)
+		sch.Required = append(sch.Required, name)
+		slices.Sort(sch.Required)
+	}
+
+	return nil
+}
+
+func parseArrayType(sch *schema.Schema, value []interface{}) error {
+	items := new(schema.Schema)
+
+	elem := value[0]
```

Comment:
should `value` error be handled here too?

Reply:
Good catch, I'll add protection for an empty array.